core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/MultiCloudWorkspaceService.scala

```diff
@@ -448,6 +448,17 @@ class MultiCloudWorkspaceService(override val ctx: RawlsRequestContext,
         )
       )
     }
+    // Necessary for DBIO.from calls
+    import dataSource.dataAccess.driver.api.DBIO
+
+    for {
+      _ <- traceDBIOWithParent("createWDSInstance", parentContext) { _ =>
```

Comment:
@davidangb Perhaps if you have a second to look....

Slightly stumped here how to conform/transform the return value here to work with `DBIO` -- see the following error. Curious to get your thoughts on what might be the best practice here -- thanks again

```
[error] /app/core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/MultiCloudWorkspaceService.scala:455:9: type mismatch;
[error]  found   : slick.dbio.DBIOAction[Unit,slick.dbio.NoStream,slick.dbio.Effect with slick.dbio.Effect]
[error]  required: scala.concurrent.Future[org.broadinstitute.dsde.rawls.model.Workspace]
[error]       _ <- traceDBIOWithParent("createWDSInstance", parentContext) { _ =>
```

Reply:
Couple comments … 
1. I'd think you want to move the call to create the WDS app up a bit, perhaps just after creation of the storage container. This would place the call inside the `for … yield` that starts at line 383. By moving the app-deployment call inside the for-yield, the return value of this entire `createWorkspace()` method reverts to being the the `yield savedWorkspace` at line 433, and you can just ignore the result of the app-deployment call.
2. Within a for-yield, each step (each statement that uses `<-`) needs to be of the same type. In this for-yield, each of these statements returns a `Future`, so I don't think you want to use `DBIO` here, you want to use `Future`. The creation of the cloud context from lines 397-402 is a good template for this.

I could chat about this live tomorrow if this isn't clear!

Reply:
This helped quite a bit!

The only change I needed to do was use the `traceWithParent` wrapper instead of the `traceDBIOWithParent` in order for the code to compile -- let me know if this is of concern